contracts/src/VouchMe.sol

```diff
@@ -18,68 +17,112 @@ contract VouchMe is ERC721URIStorage {
     // Maps token ID to testimonial details
     mapping(uint256 => Testimonial) private _testimonials;
     
-    struct Testimonial {
+    // Maps sender to receiver to their active testimonial (for one-per-pair rule)
+    mapping(address => mapping(address => uint256)) private _activeTestimonial;
+    
+    // Maps user address to their profile data
+    mapping(address => Profile) public userProfiles;    
+    struct Profile {
+        string name;
+        string contact;
+        string bio;
+    }
+      struct Testimonial {
         address sender;
         address receiver;
         string content;
+        string giverName;   
+        string profileUrl;    
         uint256 timestamp;
         bool verified;
+        bool isDeleted;
```

Comment:
Do we really need this variable? It appears to be redundant, because whenever we set it to true, we are also removing the testimonial form the list. Aren't we?